File: VB/GridControlCellEditingConfirmationExample/EditCellConfirmationBehavior.vb
Imports DevExpress.Mvvm.UI.Interactivity
Imports DevExpress.Xpf.Grid
Imports System.Windows

Namespace GridControlCellEditingConfirmationExample

    Public Class EditCellConfirmationBehavior
        Inherits Behavior(Of TableView)

        Protected Overrides Sub OnAttached()
            MyBase.OnAttached()
            AddHandler AssociatedObject.ValidateCell, AddressOf AssociatedObject_ValidateCell
        End Sub

        Private Sub AssociatedObject_ValidateCell(ByVal sender As Object, ByVal e As GridCellValidationEventArgs)
            If Equals(e.Column.FieldName, NameOf(Item.Growth)) AndAlso MessageBox.Show("Do you wish to update the value?", "Confirmation", MessageBoxButton.YesNo) = MessageBoxResult.No Then
                AssociatedObject.HideEditor()
            End If
        End Sub

        Protected Overrides Sub OnDetaching()
            RemoveHandler AssociatedObject.ValidateCell, AddressOf AssociatedObject_ValidateCell
            MyBase.OnDetaching()
        End Sub
    End Class
End Namespace

File: CS/GridControlCellEditingConfirmationExample/EditCellConfirmationBehavior.cs
using DevExpress.Mvvm.UI.Interactivity;
using DevExpress.Xpf.Grid;
using System.Windows;

namespace GridControlCellEditingConfirmationExample {
    public class EditCellConfirmationBehavior : Behavior<TableView> {
        protected override void OnAttached() {
            base.OnAttached();

            AssociatedObject.ValidateCell += AssociatedObject_ValidateCell;
        }

        private void AssociatedObject_ValidateCell(object sender, GridCellValidationEventArgs e) {
            if (e.Column.FieldName == nameof(Item.Growth) &&
                MessageBox.Show("Do you wish to update the value?", "Confirmation", MessageBoxButton.YesNo) ==
                MessageBoxResult.No) {
                AssociatedObject.HideEditor();
            }
        }

        protected override void OnDetaching() {
            AssociatedObject.ValidateCell -= AssociatedObject_ValidateCell;

            base.OnDetaching();
        }
    }
}
